packages/netlify-cms-core/src/backend.js

```diff
@@ -385,11 +385,17 @@ class Backend {
     const errors = [];
     const collectionEntriesRequests = collections
       .map(async collection => {
+        let summaryString = collection.get('summary') || '';
+        let results = summaryString ? summaryString.match(/(?<=\{\{)[^}]+(?=\}\})/g) : [];
+
         // TODO: pass search fields in as an argument
         const searchFields = [
-          selectInferedField(collection, 'title'),
-          selectInferedField(collection, 'shortTitle'),
-          selectInferedField(collection, 'author'),
+          ...new Set([
```

Comment:
Since we currently use Immutable everywhere, we should use an Immutable set here.